### DISCUSSION TITLE
How can I access the documents metadata in Milvus using Python?

### DISCUSSION THREAD

**Discussion by Danana1** [milvus-io/milvus#31206] [Q&A and General discussion]
Here is how I use Milvus so far. I know Milvus saves document metadata as well as vector info.
How can I access it? 
In particular, I want to create a system that will be able to query the existing documents and delete or update them.
If someone can refer me for such an example it will be great. This should be simple but I couldn't figure it out so far.

        #Create the collection
        my_vector_store = MilvusVectorStore(
        host = "localhost",
        port = "19530",
        dim  = self.d, 
        collection_name = "my_collection",
        index_config={"index_type": "IVF_FLAT", "nlist": 64},
        search_config={"nprobe": 16},
        overwrite=False
        )
        # Load the documents and save the embeddings
        storage_context = StorageContext.from_defaults(vector_store=self.vector_store)
        index = VectorStoreIndex.from_vector_store(self.vector_store)
        
        # Query if a document exists, it date etc... delete/update

**Comment by yhmo** [comment]
The `langchain.MilvusVectorStore` is a wrapper of milvus python sdk, some interfaces are not exposed.

You can use milvus python sdk to directly interact with milvus server.

The collection name is "my_collection". 
There are 3 default fields created by the `MilvusVectorStore`:
1. the default primary key is "pk". type is int64
2. a VARCHAR type field name "text", to store the texts that you pass to MilvusVectorStore
3. a vector field named "vector", to store the embeddings


With milvus python sdk, you can:
```
from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility,
)

collection = Collection(name=collection_name)
print(collection.describe())  # print the schema of the collection

# retrieve all documents from the collection
query_iterator = collection.query_iterator(batch_size=100, limit=1000000, expr=None, output_fields=["*"])
count = 0
while True:
    # turn to the next page
    res = query_iterator.next()
    if len(res) == 0:
        print("query iteration finished, close")
        print(f"total iterator number: {count}")
        # close the iterator
        query_iterator.close()
        break
    for i in range(len(res)):
        print(res[i])
        count = count + 1

# delete a document with primary key == 100
collection.delete(expr="pk = 100")
```
